effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -403,7 +403,15 @@ object Typer extends Phase[NameResolved, Typechecked] {
             usingCapture(capt2)
             Result(btpe, eff1)
           case _ =>
-            Context.abort(pretty"Unbox requires a boxed type, but got $vtpe")
+            Context.annotationOption(Annotations.UnboxParentDef, u) match {
+              case Some(source.DefDef(id, annot, block)) =>
+                // Since this `unbox` was synthesized by the compiler from `def foo = E`,
+                // it's possible that the user simply doesn't know that they should have used the `val` keyword to specify a value
+                // instead of using `def`; see [issue #130](https://github.com/effekt-lang/effekt/issues/130) for more details
+                Context.abort(pretty"Unbox requires a boxed type, but got $vtpe. Maybe try `val` if you're defining a value")
```

Comment:
Maybe we shouldn't talk about `unbox` anymore in this case, since it has been synthesized. What about:

"The right-hand side of a `def` binding should be a block, but got a value of type $vtpe.\nMaybe try `val` if you're defining a value."

Reply:
I used your suggestion and specified that the right-hand side of `def` is _expected_ to be a block.